### DISCUSSION TITLE
How to use a stdClass?

### DISCUSSION THREAD

**Discussion by francoism90** [livewire/livewire#5309] [1. Help] [answered]
I know `stdClass` is unsupported as a property, however I would like to know the reason why. It is still a (valid) object, right? Javascript should be able to read/write it.

Any workarounds? Why not simple add stdClass to allowed properties?

Thanks.

**Comment by Niq1982** [comment, marked as answer]
I think it's because Livewire has to initialize the property from JS on every request and it's quite hard to initialize the stdClass because we have no knowledge what the structure was earlier, since it's anonymous. So that could lead to all sorts of problems. And we already convert arrays with named keys to JS objects so that kind of does the trick. If you want to use a property so it kind of works as a object you can try to create a [Wireable property](https://laravel-livewire.com/docs/2.x/properties#wireable-properties) to carry the object in and use magic methods to update the values. 

```php
<?php
use Livewire\Wireable;

class WireableObject implements Wireable
{
    private array $data;

    public function __construct($data)
    {
        $this->data = $data;
    }

    public function __get(string $name)
    {
        if (isset($this->data[$name])) {
            return $this->data[$name];
        }

        return null;
    }

    public function __set(string $name, $value)
    {
        $this->data[$name] = $value;
    }

    public function toLivewire()
    {
        // Pass the data as key->value array to the livewire and it will then treat it as a JS Object
        return $this->data;
    }

    public static function fromLivewire($data)
    {
        return new static($data);
    }
}
```
In livewire component:

```php
class Example extends Component
{
    public WireableObject $user;
    
    public function mount() {
        $this->user = new WireableObject(['name' => 'John Smith']);
        $this->user->address = 'Testaddress 2, 12345 Test';
    }

    public function updateName()
    {
        $this->user->name = 'Lisa Smith';
    }
 }
 ```
 in blade:
 ```blade
     <div>
        Name: {{ $user->name }}<br />
        Address: {{ $user->address }}<br />

        <button wire:click="updateName" type="button">
            Update name
        </button>
    </div>
 ```

**Reply by francoism90** [reply]
@Niq1982 We did the same thing yesterday, lol! Thanks for your detailed solution. :)

It seems you can also use them as model.

**Reply by Niq1982** [reply]
Yep, you can make virtually any class or object wireable. I have recently been using Wireable with Spatie DTO:s a lot. It's fairly easy way to work with model-like data in livewire components without actually having a Eloquent model to work with.

**Reply by francoism90** [reply]
@Niq1982 Could you please give me more examples? I'm also using Spatie's DTO.
